pilot/pkg/config/aggregate/config.go

```diff
@@ -20,63 +20,66 @@ import (
 
 	"k8s.io/apimachinery/pkg/types"
 
+	"istio.io/istio/pilot/pkg/features"
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/config"
 	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/log"
 	"istio.io/istio/pkg/util/sets"
 )
 
 var errorUnsupported = errors.New("unsupported operation: the config aggregator is read-only")
 
 // makeStore creates an aggregate config store from several config stores and
 // unifies their descriptors
-func makeStore(stores []model.ConfigStore, writer model.ConfigStore) (model.ConfigStore, error) {
+func makeStore(stores []model.ConfigStore, writer model.ConfigStore) model.ConfigStore {
 	union := collection.NewSchemasBuilder()
 	storeTypes := make(map[config.GroupVersionKind][]model.ConfigStore)
 	for _, store := range stores {
 		for _, s := range store.Schemas().All() {
 			if len(storeTypes[s.GroupVersionKind()]) == 0 {
-				if err := union.Add(s); err != nil {
-					return nil, err
+				err := union.Add(s)
+				if features.EnableUnsafeAssertions && err != nil {
```

Comment:
can we not add assert every where for uncritical code path

Reply:
reverted